DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -85,19 +36,9 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
         /// <returns>Device instance if present, null if a device was not found with the provided deviceId</returns>
         public async Task<DeviceModel> GetDeviceAsync(string deviceId)
         {
-            JToken result = null;
-
-            Dictionary<string, Object> queryParams = new Dictionary<string, Object>();
-            queryParams.Add("@id", deviceId);
-            DocDbRestQueryResult response = await _docDbRestUtil.QueryCollectionAsync("SELECT VALUE root FROM root WHERE (root.DeviceProperties.DeviceID = @id)", queryParams);
-            JArray foundDevices = response.ResultSet;
-
-            if (foundDevices != null && foundDevices.Count > 0)
-            {
-                result = foundDevices.Children().ElementAt(0);
-                return result.ToObject<DeviceModel>();
-            }
-            return null;
+            var query = await _documentClient.QueryAsync();
+            var devices = query.Where(x => x.DeviceProperties.DeviceID == deviceId).ToList();
```

Comment:
I think you can use FirstOrDefault here with the condition instead of Where + FirstOrDefault